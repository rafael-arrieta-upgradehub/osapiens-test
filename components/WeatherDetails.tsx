import React from 'react';
import { View, StyleSheet } from 'react-native';
import { ThemedText } from '@/components/ThemedText';
import {IconSymbol} from "@/components/ui/IconSymbol";

interface WeatherDetailsProps {
    temperature: number;
    wind: number;
}

const WeatherDetails: React.FC<WeatherDetailsProps> = ({ temperature, wind }) => {
    return (
        <View style={styles.details}>
            <View style={styles.detailItem}>
                <IconSymbol name="temperature-high"  size={24}/>
                <ThemedText style={styles.detailText}>{`Feels like: ${(temperature - 2).toFixed(2)}Â°`}</ThemedText>
            </View>
            <View style={styles.detailItem}>
                <IconSymbol name="wind-power" size={24} />
                <ThemedText style={styles.detailText}>{`Wind: ${wind.toFixed(2)} km/h`}</ThemedText>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    details: {
        flexDirection: 'row',
        justifyContent: 'space-around',
    },
    detailItem: {
        alignItems: 'center',
    },
    detailText: {
        fontSize: 14,
        marginTop: 4,
    },
});

export default WeatherDetails;
