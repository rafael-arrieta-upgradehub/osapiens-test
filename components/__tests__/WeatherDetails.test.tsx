import React from 'react';
import renderer from 'react-test-renderer';
import WeatherDetails from '@/components/WeatherDetails';
describe('TemperatureDisplay', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<WeatherDetails temperature={25.1234} wind={10.5678} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
