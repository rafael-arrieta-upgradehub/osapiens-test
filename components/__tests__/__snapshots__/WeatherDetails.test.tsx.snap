// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureDisplay renderslñmmñlmription 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="temperature-high"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 24,
              "width": 24,
            },
            undefined,
          ],
        ]
      }
      type="monochrome"
      weight="regular"
    />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 14,
            "marginTop": 4,
          },
        ]
      }
    >
      Feels like: 23.12°
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <ViewManagerAdapter_SymbolModule
      animated={false}
      colors={[]}
      name="wind-power"
      resizeMode="scaleAspectFit"
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          [
            {
              "height": 24,
              "width": 24,
            },
            undefined,
          ],
        ]
      }
      type="monochrome"
      weight="regular"
    />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 14,
            "marginTop": 4,
          },
        ]
      }
    >
      Wind: 10.57 km/h
    </Text>
  </View>
</View>
`;
